system/jlib/jstats.cpp

```diff
@@ -439,6 +439,16 @@ StringBuffer & formatStatistic(StringBuffer & out, unsigned __int64 value, Stati
     return formatStatistic(out, value, queryMeasure(kind));
 }
 
+constexpr static stat_type usTimestampThreshold = 100ULL * 365ULL * 24ULL * 3600'000000ULL; // 100 years in us
+// Support the gradual conversion of timestamps from us to ns.  If a timestamp is in the old format (before 2070)
+// then it is assumed to be in us, otherwise ns.
+stat_type normalizeTimestampToNs(stat_type value)
```

Comment:
Function name should specify the expected incoming unit of measurement...
normalizeTimestampUsToNs ?

param name should be self documenting as well...
timestampNs?

Reply:
It will accept either Ns or Us, which is why the incoming unit is not specified.